tests/loader_debug_ext_tests.cpp

```diff
@@ -0,0 +1,934 @@
+ * deal in the Materials without restriction, including without limitation the
+ * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+ * sell copies of the Materials, and to permit persons to whom the Materials are
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice(s) and this permission notice shall be included in
+ * all copies or substantial portions of the Materials.
+ *
+ * THE MATERIALS ARE PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ *
+ * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE MATERIALS OR THE
+ * USE OR OTHER DEALINGS IN THE MATERIALS.
+ *
+ * Author: Charles Giessen <charles@lunarg.com>
+ * Author: Mark Young <marky@lunarg.com>
+ */
+
+#include "test_environment.h"
+
+//
+// VK_EXT_debug_report specific tests
+// =========================================
+//
+
+// Prototype declaration for callback so we can use it in class utility methods
+static VkBool32 VKAPI_CALL test_DebugReportCallback(VkDebugReportFlagsEXT flags, VkDebugReportObjectTypeEXT objectType,
+                                                    uint64_t object, size_t location, int32_t messageCode, const char* pLayerPrefix,
+                                                    const char* pMessage, void* pUserData);
+
+class DebugReportTest : public ::testing::Test {
+   public:
+    void VerifyExpected(VkDebugReportFlagsEXT flags, VkDebugReportObjectTypeEXT object_type, const std::string& message) {
+        if (object_type == expected_object_type && 0 < (flags | expected_flag)) {
+            if (allow_any_message || (std::string::npos != message.find(expected_message))) {
+                message_found = true;
+            }
+        }
+    }
+
+   protected:
+    virtual void SetUp() {
+        env = std::unique_ptr<FrameworkEnvironment>(new FrameworkEnvironment());
+        for (uint32_t icd = 0; icd < 3; ++icd) {
+            env->add_icd(TestICDDetails(TEST_ICD_PATH_VERSION_6, VK_API_VERSION_1_0));
+            env->get_test_icd(icd).physical_devices.push_back({});
+            env->get_test_icd(icd).physical_devices.push_back({});
+        }
+        // Initialize the expected output
+        allow_any_message = false;
+        expected_message = "";
+        expected_object_type = VK_DEBUG_REPORT_OBJECT_TYPE_UNKNOWN_EXT;
+        message_found = false;
+    }
+
+    VkResult CreateReportInstance(VkDebugReportFlagsEXT debug_report_flags, VkInstance* inst,
+                                  VkApplicationInfo* app_info = nullptr) {
+        static std::vector<const char*> enabled_extensions;
+        enabled_extensions.push_back(VK_EXT_DEBUG_REPORT_EXTENSION_NAME);
```

Comment:
Wouldn't this keep adding the extension over and over again? As static is initialized once but this code is run many many times. 
It could be put in the `SetUp()` function since its the same every time.

Reply:
Dang, good point.  No, I like it here just so it's all together.  The static was wrong and a left-over.  Fixed.